addons/account/models/account_reconcile_model.py

```diff
@@ -999,7 +999,9 @@ def _check_rule_propositions(self, st_line, amls_values_list):
             for aml_values in amls_values_list
         )
         sign = 1 if st_line_amount_curr > 0.0 else -1
-        amount_curr_after_rec = sign * (amls_amount_curr + st_line_amount_curr)
+        amount_curr_after_rec = st_line_currency.round(
```

Comment:
@smetl I'm not sure if I should take `st_line_currency` or `aml_vales['aml'].currency_id`, same for lines 1021 and 1026